frontend/packages/core-api-client/src/client/Record.ts

```diff
@@ -1,34 +1,125 @@
+import { getFieldKind } from '..';
 import {
+  Record,
+  FieldKind,
   RecordCreateRequest,
   RecordCreateResponse,
   RecordGetRequest,
   RecordGetResponse,
   RecordListRequest,
   RecordListResponse,
+  RecordList,
 } from '../types';
 
 import { BaseRESTClient } from './BaseRESTClient';
+import { FormClient } from './Form';
 
 export class RecordClient {
   restClient: BaseRESTClient;
 
-  constructor(restClient: BaseRESTClient) {
+  formClient: FormClient;
+
+  constructor(restClient: BaseRESTClient, formClient: FormClient) {
     this.restClient = restClient;
+    this.formClient = formClient;
   }
 
   create = (request: RecordCreateRequest): Promise<RecordCreateResponse> => {
     return this.restClient.do(request, '/records', 'post', request.object, 200);
   };
 
-  list = (request: RecordListRequest): Promise<RecordListResponse> => {
+  list = async (request: RecordListRequest): Promise<RecordListResponse> => {
```

Comment:
maybe we should think about doing this in chunks, this could take a long time to finish if there's a lot of records with a lot of subrecords

Reply:
Yeah, we could even pass an arg in to not to do it. When we're listing records we typically won't need all of the subrecord values. I'll leave it as is for now, but something to look at when someone works on the recipients list ticket.